paper-api/src/main/java/io/papermc/paper/world/damagesource/CombatEntry.java

```diff
@@ -0,0 +1,79 @@
+package io.papermc.paper.world.damagesource;
+
+import io.papermc.paper.InternalAPIBridge;
+import org.bukkit.damage.DamageSource;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.LivingEntity;
+import org.jetbrains.annotations.ApiStatus;
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+
+/**
+ * Represents a combat entry
+ *
+ * @since 1.21.4
```

Comment:
We have for now decided against since annotation in the project.
Adding *some* is just gonna make our lifes even more miserable when/if we solve them semi automatically.